src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1305,13 +1306,204 @@ public static Matrix4x4 CreateReflection(Plane value)
+            V12 = Permute(V12, 0x2C);        //_MM_SHUFFLE(0, 2, 3, 0)
+            V03 = Permute(row3, 0x8D);       //_MM_SHUFFLE(2, 0, 3, 1)
+
+            // V13 = D1X,D1W,D2Z,D2W
+            V13 = Sse.Shuffle(D1, D2, 0xEC); //_MM_SHUFFLE(3, 2, 3, 0)
+            V13 = Permute(V13, 0x93);        //_MM_SHUFFLE(2, 1, 0, 3)
+
+            V00 = Sse.Multiply(V00, V10);
+            V01 = Sse.Multiply(V01, V11);
+            V02 = Sse.Multiply(V02, V12);
+            V03 = Sse.Multiply(V03, V13);
+
+            Vector128<float> C1 = Sse.Subtract(C0, V00);
+            C0 = Sse.Add(C0, V00);
+            Vector128<float> C3 = Sse.Add(C2, V01);
+            C2 = Sse.Subtract(C2, V01);
+            Vector128<float> C5 = Sse.Subtract(C4, V02);
+            C4 = Sse.Add(C4, V02);
+            Vector128<float> C7 = Sse.Add(C6, V03);
+            C6 = Sse.Subtract(C6, V03);
+
+            C0 = Sse.Shuffle(C0, C1, 0xD8); //_MM_SHUFFLE(3, 1, 2, 0)
+            C2 = Sse.Shuffle(C2, C3, 0xD8); //_MM_SHUFFLE(3, 1, 2, 0)
+            C4 = Sse.Shuffle(C4, C5, 0xD8); //_MM_SHUFFLE(3, 1, 2, 0)
+            C6 = Sse.Shuffle(C6, C7, 0xD8); //_MM_SHUFFLE(3, 1, 2, 0)
+
+            C0 = Permute(C0, 0xD8); //_MM_SHUFFLE(3, 1, 2, 0)
+            C2 = Permute(C2, 0xD8); //_MM_SHUFFLE(3, 1, 2, 0)
+            C4 = Permute(C4, 0xD8); //_MM_SHUFFLE(3, 1, 2, 0)
+            C6 = Permute(C6, 0xD8); //_MM_SHUFFLE(3, 1, 2, 0)
+
+            // Get the determinant
+            vTemp2 = row1;
+            var det = Vector4.Dot(C0.AsVector4(), vTemp2.AsVector4());
+
+            // Check determinate is not zero
+            if (MathF.Abs(det) < float.Epsilon)
+            {
+                result = new Matrix4x4(float.NaN, float.NaN, float.NaN, float.NaN,
+                            float.NaN, float.NaN, float.NaN, float.NaN,
+                            float.NaN, float.NaN, float.NaN, float.NaN,
+                            float.NaN, float.NaN, float.NaN, float.NaN);
+                return false;
+            }
+
+            // Create Vector128<float> copy of the determinant and invert them.
+            Vector128<float> vTemp = Vector128.Create(det, det, det, det);
+            Vector128<float> ones = Vector128.Create(1.0f, 1.0f, 1.0f, 1.0f);
+            vTemp = Sse.Divide(ones, vTemp);
+
+            row1 = Sse.Multiply(C0, vTemp);
+            row2 = Sse.Multiply(C2, vTemp);
+            row3 = Sse.Multiply(C4, vTemp);
+            row4 = Sse.Multiply(C6, vTemp);
+
+            Unsafe.SkipInit(out result);
+            ref Vector128<float> vResult = ref Unsafe.As<Matrix4x4, Vector128<float>>(ref result);
+
+            vResult = row1;
+            Unsafe.Add(ref vResult, 1) = row2;
+            Unsafe.Add(ref vResult, 2) = row3;
+            Unsafe.Add(ref vResult, 3) = row4;
+
+            return true;
+        }
+
+        public static bool SoftwareFallback(Matrix4x4 matrix, out Matrix4x4 result)
```

Comment:
This can't be public and both should likely be `static local functions` as they will only ever be used from within the publicly exposed API.